src/evict/evict_page.c

```diff
@@ -213,6 +213,21 @@ __wt_evict(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t previous_state, uint3
 
     if (0) {
 err:
+        /*
+         * Add back the eviction failed page to the urgent queue, if the page is from history store.
+         * It is possible that checkpoint adds an obsolete page of history store to get it evicted
+         * but because of concurrent access of the page by checkpoint and eviction, it may fail. Add
+         * them back to the queue and try to evict again.
+         *
+         * It is possible that an eviction failure occurs to a page that is not added by a
+         * checkpoint thread, but we consider those scenarios are limited.
+         */
+        if (WT_IS_LAS(S2BT(session))) {
+            WT_IGNORE_RET_BOOL(__wt_page_evict_urgent(session, ref));
+            WT_STAT_CONN_INCR(session, cache_eviction_history_store_pages_queued_urgent_again);
+            __wt_yield();
+        }
```

Comment:
My opinion is we should ignore these failures entirely.

As far as I know, we don't have any data that indicates this work is necessary, and eviction will get around to these pages. I don't yet have any reason to believe it's going to be urgent for us to handle these pages, so I would prefer to not complicate the code absent evidence it's necessary.

Reply:
The eviction server chooses the trees randomly to evict the pages under them.  Due to randomness, eviction may not evict the obsolete history pages and evict the pages that are may actually require. This may lead to cache use by pages that are not required.

If I am the only one concern about this scenario, I am fine to ignore this problem.
Another approach of how about choosing `LAS` as the first dhandle by the eviction walk when considering the pages to be evicted?

@agorrod, @bvpvamsikrishna Any suggestions?